be/src/runtime/task_group/task_group.h

```diff
@@ -35,14 +35,15 @@ class PipelineTask;
 }
 
 class TPipelineResourceGroup;
+class MemTrackerLimiter;
+struct TrackerLimiterGroup;
+using TrackerLimiterGroups = std::vector<TrackerLimiterGroup>;
```

Comment:
A query memtracker can be both held by TaskGroups memtacker list and Global's mem_tracker_limiter_pool, so I think here should be ```std::vector<shared_ptr<TrackerLimiterGroup>>```;